xarray/backends/zarr.py

```diff
@@ -816,11 +816,29 @@ def open_store_variable(self, name):
             if zarr_array.fill_value is not None:
                 attributes["_FillValue"] = zarr_array.fill_value
         elif "_FillValue" in attributes:
-            attributes["_FillValue"] = FillValueCoder.decode(
-                attributes["_FillValue"], zarr_array.dtype
-            )
+            # TODO update version check for the released version with dtypes
+            #  probably be 3.1
+
+            # if Version(zarr.__version__) > Version("3.0.10"):
+            if hasattr(zarr_array.metadata.data_type, "to_native_dtype"):
```

Comment:
Had to check by attrs here because of version confusion. Though maybe this is a better check here anyway.

Reply:
better check than version comparison*

Reply:
This is a good callout, I'm a little surprised the tests didn't pick that up. 

For merging we can do >=3.1.0 i think, but I ran into problems testing with version checking because the release of 3.0.10 is higher than 3.0.9.dev (and for some reason the instlalation from git was not turning it into 3.0.10.dev)